### ISSUE TITLE
[BUG] device custom sounds doesn't work

### ISSUE THREAD

**Issue by aaronvine** [twilio/twilio-client.js#113]
- [x] I have verified that the issue occurs with the latest twilio.js release and is not marked as a known issue in the [CHANGELOG.md](https://github.com/twilio/twilio-client.js/blob/master/CHANGELOG.md).
 - [x] I reviewed the [Common Issues](https://github.com/twilio/twilio-client.js/blob/master/COMMON_ISSUES.md) and open GitHub issues and verified that this report represents a potentially new issue.
 - [x] I verified that the [Quickstart](https://github.com/TwilioDevEd/client-quickstart-js) application works in my environment.
 - [x] I am not sharing any [Personally Identifiable Information (PII)](https://www.twilio.com/docs/glossary/what-is-personally-identifiable-information-pii)
       or sensitive account information (API keys, credentials, etc.) when reporting this issue.

We've been trying to override current incoming sound behavior - making the incoming sound play only once without repeat.
After checking the docs, we've passed the `sounds` object multiple ways but it doesn't seem to work.
When we pass the `sounds` object, the default behavior is overridden but we don't hear any sound at all.

**Code to reproduce the issue:**

```js
device.setup(token, {
  debug,
  sounds: {
    incoming: {
      filename: 'https://media.twiliocdn.com/sdk/js/client/sounds/releases/1.0.0/incoming.mp3',
      shouldLoop: false,
    },
  },
});
```

We've also tried passing it in a different way but it also didn't work:
```js
device.setup(token, {
  debug,
  sounds: {
    incoming: 'https://media.twiliocdn.com/sdk/js/client/sounds/releases/1.0.0/incoming.mp3',
  },
});
```

**Expected behavior:**
Incoming sound is overridden, playing sound from url one, no repeat

**Actual behavior:**
No incoming sounds at all

Also, it would be great to have an option to also set `maxDuration` for the `incoming` sound property as the others

Thanks

**Comment by charliesantos** [comment]
Hi @aaronvine , thank you for reporting. I can confirm that the following code works
```
device.setup(token, {
  debug,
  sounds: {
    incoming: 'https://api.twilio.com/cowbell.mp3',
  }
});
```
Also, we continue to play incoming sound as long as the incoming call is pending as we've mentioned in the docs. 

If you are not able to hear incoming sound, please make sure CORS is enabled on your server.

If you'd like to only play the incoming sound once, as a workaround, you can disable the incoming sound and play your own sound in response to an incoming event.

Thank you,
Charlie

**Comment by aaronvine** [comment]
@charliesantos thanks
I have tried the url in your example (https://api.twilio.com/cowbell.mp3) and indeed it works
However, when trying other sounds/urls (https://media.twiliocdn.com/sdk/js/client/sounds/releases/1.0.0/incoming.mp3 for example) it doesn't seem to work.
Are there any restrictions on the sound file? size/duration/url?

Thanks

**Comment by cybex-dev** [comment]
> Hi @aaronvine , thank you for reporting. I can confirm that the following code works
> 
> ```
> device.setup(token, {
>   debug,
>   sounds: {
>     incoming: 'https://api.twilio.com/cowbell.mp3',
>   }
> });
> ```
> 
> Also, we continue to play incoming sound as long as the incoming call is pending as we've mentioned in the docs.
> 
> If you are not able to hear incoming sound, please make sure CORS is enabled on your server.
> 
> If you'd like to only play the incoming sound once, as a workaround, you can disable the incoming sound and play your own sound in response to an incoming event.
> 
> Thank you, Charlie

For the record, anyone coming across this answer, you'll have to use: 
```ts
const options = {
    sounds: { ... },
}
const device = new Device(token, options)
```

See [documentation](https://www.twilio.com/docs/voice/sdks/javascript/twiliodevice#instantiate-a-device-with-deviceoptions) for more information.
